### PR TITLE
add feature varnet

### PR DESCRIPTION
**Pull request by GiannakopoulosIlias**
Added a new folder in fastmri_examples which includes the new modules for feature varent and ablations.
Added a new model file (feature_varnet) in fastmri/models which includes the new models.
Updated markdowns to include the new reference.

### CODE REVIEW

File: fastmri/models/feature_varnet.py [hunk 1]
```diff
@@ -0,0 +1,1575 @@
+from typing import NamedTuple, Optional, Tuple, List
```
Comment (mmuckley) [inline fastmri/models/feature_varnet.py:1]: Need the copyright header here - see other files in repo.

File: fastmri_examples/feature_varnet/pl_modules/feature_varnet_module.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
+import math
```
Comment (mmuckley) [inline fastmri_examples/feature_varnet/pl_modules/feature_varnet_module.py:1]: Add copyright header.

File: fastmri_examples/feature_varnet/train_feature_varnet.py [hunk 3]
```diff
@@ -0,0 +1,270 @@
+import os
```
Comment (mmuckley) [inline fastmri_examples/feature_varnet/train_feature_varnet.py:1]: Add copyright header.

File: fastmri_examples/feature_varnet/pl_modules/__init__.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (mmuckley) [inline fastmri_examples/feature_varnet/pl_modules/__init__.py:1]: This folder should have a different name and potentially gives a namespace conflict. Could you move `feature_varnet_module.py` to the top level and remove this folder? Then in `train_feature_varnet.py` import the module from `feature_varnet_module.py`.

**Review by mmuckley — CHANGES_REQUESTED** [review]
Mostly asking for small reformatting items.

**Comment by mmuckley** [comment]
Also @GiannakopoulosIlias it looks like the PR is failing some linter tests. Could you also fix those?

**Comment by mmuckley** [comment]
@GiannakopoulosIlias some tests still aren't passing. Please make sure you're using the correct version of `black` on your code.

**Comment by mmuckley** [comment]
@GiannakopoulosIlias the message about Jupyter can be ignored. You only need to `pip install black` with the right version listed in the message.

**Comment by GiannakopoulosIlias** [comment]
> @GiannakopoulosIlias the message about Jupyter can be ignored. You only need to `pip install black` with the right version listed in the message.

Should be okay now, thanks!

**Comment by mmuckley** [comment]
Hello @GiannakopoulosIlias, now it looks like there are errors with type annotations with mypy.

If there are too many to deal with, you could move `feature_varnet.py` into the examples folder. Then you should be able to commit the code without fixing everything.

**Comment by GiannakopoulosIlias** [comment]
> Hello @GiannakopoulosIlias, now it looks like there are errors with type annotations with mypy.
> 
> If there are too many to deal with, you could move `feature_varnet.py` into the examples folder. Then you should be able to commit the code without fixing everything.

I will try to adjust them tomorrow. They look many but I think most of them are the same error for all different model ablations.

**Review by mmuckley — APPROVED** [review]
Approving with fixes for moving the `feature_varnet.py` to `fastmri_examples`.

**Pull request merged**